ecs-cli/modules/utils/compose/convert_task_definition.go

```diff
@@ -775,19 +780,21 @@ func ConvertToVolumes(volumeConfigs map[string]*config.VolumeConfig) *volumes {
 	if volumeConfigs != nil {
 		for name, config := range volumeConfigs {
 			if config != nil {
+				// NOTE: If Driver field is not empty, this
+				// will add a prefix to the named volume on the container
 				if config.Driver != "" {
-					log.WithFields(log.Fields{
-						"volume name": name,
-						"option name": "driver",
-					}).Warn("Skipping unsupported YAML option...")
+					return nil, errors.New("Volume driver is not supported")
 				}
+				// Driver Options must relate to a specific volume driver
 				if len(config.DriverOpts) != 0 {
-					log.WithFields(log.Fields{
-						"volume name": name,
-						"option name": "driver_opts",
-					}).Warn("Skipping unsupported YAML option...")
+					return nil, errors.New("Volume driver options is not supported")
 				}
+				// NOTE: explicitly setting external to false
+				// will cause compose up to fail, but if not
```

Comment:
Missing a negative here? "will not cause"? It won't cause it to fail right?

Reply:
No, it will:
## Fails if external is false
```
~/ecs/ecs-cli/testing/projects/test_fields$ ecs-cli compose up --cluster foobar --region us-east-2
ERRO[0000] Unable to open ECS Compose Project            error="named volume [testfields_foo:/var/lib/backup/data] is used but no declaration was found in the volumes section"
FATA[0000] Unable to create and read ECS Compose Project  error="named volume [testfields_foo:/var/lib/backup/data] is used but no declaration was found in the volumes section"

~/ecs/ecs-cli/testing/projects/test_fields$ cat docker-compose.yml
version: '2'
services:
  wordpress:
    image: wordpress
    ports:
      - "80:80"
  mysql:
    image: mysql
    environment:
      MYSQL_ROOT_PASSWORD: foobar
    volumes:
      - foo:/var/lib/backup/data
volumes:
  foo:
    external: false
```

Reply:
You bring up a good point though-- this is confusing and the test is not accurate. Fixing.

Reply:
So the issue was that `external: true` would succeed and `external: false` would fail when running compose up. That suggests that we should only return an error when `config.External.External` was false; however that was causing the tests to fail because we were instantiating empty `VolumeConfig` objects, which populated the fields to the default nil value -- which in the case of the `External` field was false. In reality, however, if there is no `external` subfield in a docker-compose file (it is mutually exclusive with `driver` and `driver_opts`) libcompose parses it as nil, rather than an empty struct. 

The fix was to update the tests to set the value for a named volume to be `nil` rather than an empty `config.VolumeConfig` struct. The code now returns an error if any volume subfield is specified, regardless of value.